libminifi/src/utils/StringUtils.cpp

```diff
@@ -35,7 +35,11 @@ std::string StringUtils::trim(std::string s) {
   return trimRight(trimLeft(s));
 }
 
-std::vector<std::string> StringUtils::split(const std::string &str, const std::string &delimiter) {
```

Comment:
Line 43 here could have caused bad memory access in case an empty delimiter. This was a potential SPoF in the `ExpressionLanguage`'s `allDelineatedValues`.